util/mods.py

```diff
@@ -776,24 +991,26 @@ def add_incompatibility_to_mod(
         # If the value is a single string...
         if isinstance(dependency_or_dependency_ids, str):
             dependency_id = dependency_or_dependency_ids.lower()
-            if dependency_id in all_mods:
-                mod_data["incompatibilities"].add(dependency_id)
-                # if "incompatibilities" not in all_mods[dependency_id]:
-                #     all_mods[dependency_id]["incompatibilities"] = set()
-                # all_mods[dependency_id]["incompatibilities"].add(mod_data["packageId"])
+            for uuid in all_mods:
+                if all_mods[uuid]["packageId"] == dependency_id:
+                    mod_data["incompatibilities"].add(dependency_id)
+                    # if "incompatibilities" not in all_mods[dependency_id]:
+                    #     all_mods[dependency_id]["incompatibilities"] = set()
+                    # all_mods[dependency_id]["incompatibilities"].add(mod_data["packageId"])
```

Comment:
You can definitely remove these comments if you want

Reply:
Fixed